Packs/GCP-Enrichment-Remediation/Scripts/GCPOffendingFirewallRule/GCPOffendingFirewallRule.py

```diff
@@ -0,0 +1,128 @@
+    return False
+
+
+def test_match(port: str, protocol: str, rule: Dict, network_tags: list) -> bool:
+    """
+    Breaks a string port range (i.e. '20-25') into integers for comapirson
+    Args:
+        port (str): port supplied in script args.
+        protocol (str): protocol supplied in script args.
+        rule (Dict): GCP firewall rule pulled from integration command.
+        network_tags (list): list of network tags (can be empty).
+
+    Returns:
+        bool: whether there was a match between the supplied port, protocol
+        and possible target tag combination and the GCP firewall rule.
+    """
+    # Match rule needs to be direction ingress, source from internet (0.0.0.0/0), enabled and an allow rule.
+    if rule.get('direction') == 'INGRESS' and '0.0.0.0/0' in rule.get('sourceRanges', []) \
+       and rule.get('disabled') is False and 'allowed' in rule.keys():
+        # Test if targetTags are relevant or not (if show up in keys or tag match)
+        target_tags_verdict = ('targetTags' not in rule.keys() or len(set(rule.get('targetTags', [])) & set(network_tags)) > 0)
+        for entry in rule.get('allowed', []):
+            # Match is all protocol AND either no target tags OR target tags match
+            if entry.get('IPProtocol') == 'all' and target_tags_verdict:
+                return True
+            # Complicated because just {'IPProtocol': 'udp'} means all udp ports
+            # therefore if protocol match but no ports, this is a match
+            elif entry.get('IPProtocol') == protocol.lower() and 'ports' not in entry.keys():
+                return True
+            # Else need to go through all ports to see if range or not
+            elif entry.get('IPProtocol') == protocol.lower() and 'ports' in entry.keys():
+                for port_entry in entry.get('ports'):
+                    if "-" in port_entry:
+                        res = test_range(port_entry, port)
+                        if res and target_tags_verdict:
+                            return True
+                    else:
+                        if port == port_entry and target_tags_verdict:
+                            return True
+    return False
+
+
+''' COMMAND FUNCTION '''
+
+
+def gcp_offending_firewall_rule(args: Dict[str, Any]) -> CommandResults:
+    """
+    Determine potential offending firewall rules in GCP based on port, protocol and possibly target tags (network tags).
+    Args:
+        args (dict): Command arguments from XSOAR.
+
+    Returns:
+        list[CommandResults]: outputs, readable outputs and raw response for XSOAR.
+
+    """
+
+    project_id = args.get("project_id")
+    network_url = args.get("network_url")
+    port = args.get("port", "NO PORT")
+    protocol = args.get("protocol", "NO PROTOCOL")
+    network_tags = args.get("network_tags", [])
+
+    # Using `demisto.executeCommand` instead of `execute_command` because for
+    # multiple integration instances we can expect one too error out.
```

Comment:
So we don't want to raise an error?
```python
    if isError(fw_rules):
        raise Exception(get_error(fw_rules))

```

Reply:
I have this on other automations as well that we don't want to ever fail: https://github.com/demisto/content/blob/d2b6f5f6cd8472a31535b7e1da1f252debcce144/Packs/AWS-Enrichment-Remediation/Scripts/AWSRecreateSG/AWSRecreateSG.py#L298